app/generic/proof/proof.py

```diff
@@ -0,0 +1,217 @@
+import logging
+from enum import Enum
+from typing import Literal, Optional
+
+from aries_cloudcontroller import (
+    AcaPyClient,
+    IndyPresSpec,
+    IndyProofRequest,
+    V10PresentationSendRequestRequest,
+    V20PresRequestByFormat,
+    V20PresSendRequestRequest,
+    V20PresSpecByFormatRequest,
+)
+from fastapi import APIRouter, Depends
+
+from app.dependencies import agent_selector
+from app.generic.proof.facades.acapy_proof_v1 import ProofsV1
+from app.generic.proof.facades.acapy_proof_v2 import ProofsV2
+from app.generic.proof.models import PresentationExchange
+
+logger = logging.getLogger(__name__)
+
+
+router = APIRouter(prefix="/generic/proof", tags=["proof"])
+
+ProtocolVersion = Literal["1", "2"]
+
+
+class ProofsFacade(Enum):
+    v10 = ProofsV1
+    v20 = ProofsV2
+
+
+@router.post("/send-request")
+async def send_proof_request(
+    connection_id: str,
+    proof_request: IndyProofRequest,
+    protocol_version: Optional[ProtocolVersion] = "1",
+    aries_controller: AcaPyClient = Depends(agent_selector),
```

Comment:
Please see the api of the issuer endpoints. I integrated the connection id and protocol version into the body of the request.

This method is also fine, but I think above all we should be consistent in how the API looks. 

```
POST /generic/issuer/credentials
```

vs

```
POST /generic/proof/send-request?protocol_version=2&connection_id=xxxx
```